### PR TITLE
Add #96264: Checkout git with detach flag

### LINKED ISSUE #96264: Option to disable creation of a branch when a remote tracking branch is selected
When using vscode with Git repositories, in the branch switching pane if a remote tracking branch is selected, vscode automatically creates a local branch using selected remote tracking branch.

It would be useful to have a setting to turn this off and checkout the remote tracking branch in a detached state. I use branches in detached state to browse the code, run quick builds, copy code. With the local branches getting created, my repo ends up with a lot of those and I end up spending time later to cleanup the local branches. 

To get around this, I switch to the terminal and checkout the remote tracking branch (`git checkout origin/<remote_branch_name>`) and then switch back to vscode.

### PR DESCRIPTION
**Pull request by markusheinemann**
This PR fixes  #96264

### Description
This PR allows the user to checkout from a remote branch with a detached head. It will add `config.checkoutDetached` as a setting to the default git extension. If this option is set to true, the `--detach` flag will be added to the git checkout command. 

### How to Test
To test this feature you'll have to checkout a remote branch from the git menu. If the setting is enabled the new branch should have a detached head.

### CODE REVIEW

**Review by bharath144 — APPROVED** [review]

**Comment by heinemannmarkus** [comment]
Anything new here?

**Comment by bharath-m-g** [comment]
Hey @alexr00 any clue on when this will get merged. PRs been open for months now.

File: extensions/git/src/git.ts [hunk 1]
```diff
@@ -1287,6 +1288,10 @@ export class Repository {
 			args.push(treeish);
 		}
 
+		if (config.get('checkoutDetached')) {
+			args.push('--detach');
```
Comment (ssergiienko) [inline extensions/git/src/git.ts:1292]: Plz, correct me, but from what I see, you are using both `--track` and `--detach` options simultaneously, which is not allowed:

```
LANG=C git checkout -q --detach -t origin/master
fatal: '--detach' cannot be used with '-b/-B/--orphan'

git --version
git version 2.28.0
```

Am I missing something?

Comment (markuscodes1) [reply extensions/git/src/git.ts:1302]: Sorry, my mistake. Should be fixed with the last commit.

**Comment by joaomoreno** [comment]
Sorry, but when reviewing the PR I concluded that a setting is not great for this. We can simply drive the user to checkout in detached mode instead, so I've implemented that:

recording (8)

Thanks anyway!

**Pull request closed without merging**